go-controller/pkg/util/kube.go

```diff
@@ -265,6 +265,11 @@ func PodScheduled(pod *kapi.Pod) bool {
 	return pod.Spec.NodeName != ""
 }
 
+// PodTerminatingWithFinalizer checks if the pod has been deleted via API but still exists due to a finalizer
+func PodTerminatingWithFinalizer(pod *kapi.Pod) bool {
```

Comment:
NIT: If all we will do is to look at `DeletionTimestamp` -- which I have no problems with -- maybe this functions should be called `PodTerminating` or `PodDeleted` ?

Reply:
Question: `DeletionTimestamp` is an attribute that a regular user is not able to set, right? I'm thinking of pathological cases where k8 is fed a yaml file for a pod, where annotations and this attribute is already set for some reason.

Reply:
Awesome!!!!!